comm/src/main/scala/coop/rchain/comm/discovery/GrpcKademliaRPC.scala

```diff
@@ -0,0 +1,109 @@
+package coop.rchain.comm.discovery
+
+import cats._, cats.data._, cats.implicits._, cats.mtl._
+import coop.rchain.catscontrib._, Catscontrib._, ski._, TaskContrib._
+import coop.rchain.comm._
+import monix.eval._
+import monix.execution.atomic._
+import monix.execution._
+import coop.rchain.metrics.Metrics
+import coop.rchain.shared.{Cell, Log, LogSource}
+import scala.concurrent.Future
+import io.grpc._, io.grpc.netty._
+import com.google.protobuf.ByteString
+import scala.concurrent.duration._
+import com.google.protobuf.ByteString
+
+class GrpcKademliaRPC(src: PeerNode, port: Int, timeout: FiniteDuration)(implicit
+                                                                         scheduler: Scheduler,
+                                                                         metrics: Metrics[Task],
+                                                                         log: Log[Task])
+    extends KademliaRPC[Task] {
+
+  private implicit val logSource: LogSource = LogSource(this.getClass)
+
+  def ping(peer: PeerNode): Task[Boolean] =
+    for {
+      _       <- Metrics[Task].incrementCounter("protocol-ping-sends")
+      channel <- clientChannel(peer)
+      pongErr <- Task
+                  .fromFuture {
```

Comment:
Is there a reason why you don't use grpc-monix?

Reply:
next PR